rsocket/RSocketConnectionEvents.h

```diff
@@ -8,12 +8,35 @@ class exception_wrapper;
 
 namespace rsocket {
 
+// The application should implement this interface to get called-back
+// on network events.
 class RSocketConnectionEvents {
  public:
   virtual ~RSocketConnectionEvents() = default;
 
+  // This method gets called when the underlying transport is connected to the
+  // remote side.  This does not necessarily mean that the RSocket connection
+  // will be successful.  As an example, the transport might get reconnected
+  // for an existing RSocketStateMachine.  But the resumption at RSocket layer
```

Comment:
Nit: `But the resumption at RSocket layer` --> `But resumption at the RSocket layer`